diff --git a/src/libbluray/bdj/bdj.c b/src/libbluray/bdj/bdj.c
index 359bc8f..32b080c 100644
--- a/src/libbluray/bdj/bdj.c
+++ b/src/libbluray/bdj/bdj.c
@@ -256,7 +256,7 @@ static char *_java_home_macos()
         return NULL;
     }
 
-    switch (java_home_pid = vfork())
+    switch (java_home_pid = -1)
     {
         case -1:
             BD_DEBUG(DBG_BDJ | DBG_CRIT, "vfork failed\n");
@@ -270,7 +270,7 @@ static char *_java_home_macos()
             close(fd[1]);
             close(fd[0]);
 
-            execl(MACOS_JAVA_HOME, MACOS_JAVA_HOME);
+            //execl(MACOS_JAVA_HOME, MACOS_JAVA_HOME);
 
             _exit(-1);
 
